### PR TITLE
Handle many files as a zip archive

### PR DESCRIPTION
**Pull request by lf94**
Resolves https://github.com/KittyCAD/modeling-app/issues/1151

### CODE REVIEW

**Comment by lf94** [comment]
Best option it is :wink:

**Comment by lf94** [comment]
Requires a small patch to `cli`. I will notify when it's merged.

**Comment by lf94** [comment]
I think I have to go a bit deeper than `cli` and add a patch to `format` for proper `buffers: { uri }` data.

**Comment by lf94** [comment]
Awaiting https://github.com/KittyCAD/cli/pull/527

**Review by Irev-Dev — APPROVED** [review]

**Review by Irev-Dev — COMMENTED** [review]
Once https://github.com/KittyCAD/cli/pull/527 is ready.

We'll need to update https://github.com/KittyCAD/action-install-cli
Basically just updating the version and hashes like:
https://github.com/KittyCAD/action-install-cli/pull/15/files
Using the hashes from the cli release notes.

And there's`- uses: KittyCAD/action-install-cli@v0.2.21` in `playwright.yml`

Then hopefully it's as easy as
```
// TODO: gltfs don't seem to work with snap shots. push onto exportLocations once it's figured out
  await doExport({
    type: 'gltf',
    storage: 'standard',
    presentation: 'pretty',
  })
```
To
```
exportLocations.push(
    await doExport({
      type: 'gltf',
      storage: 'standard',
      presentation: 'pretty',
    })
  )
```

And updating

**Comment by lf94** [comment]
Yeah I've already made that change in the tests; I'll just fire a re-run after the changes in the other repo are done :)

**Comment by Irev-Dev** [comment]
I just remembered one more thing, the reason why the gltf is still being committed is because of gitignore so that will need to be edited too.

Currently there are these lines in the file
```gitignore
e2e/playwright/export-snapshots/*.ply
e2e/playwright/export-snapshots/*.obj
e2e/playwright/export-snapshots/*.step
e2e/playwright/export-snapshots/*.stl
e2e/playwright/export-snapshots/*binary.gltf
e2e/playwright/export-snapshots/*embedded.gltf
```
so you could add `e2e/playwright/export-snapshots/*standard.gltf`

or condense all of them to just
```gitignore
e2e/playwright/export-snapshots/*
!e2e/playwright/export-snapshots/*.png
```
 since there are only pngs in there now we want to keep.
 
 When you do so, I think you'll need to delete the two `.gltf` files that are left as clean-up.

**Comment by Irev-Dev** [comment]
Sorry this ended this issue's been dragging one.

**Comment by Irev-Dev** [comment]
Sorry I made the change instead of talking about it.

**Comment by lf94** [comment]
I don't see it that way at all! It's really no problem! :)

**Comment by lf94** [comment]
I need to modify the test so it unzips the newly created output.zip that holds the gltf files.

**Comment by lf94** [comment]
@Irev-Dev could you give one last look before we merge?

File: output.bin
Comment (Irev-Dev) [inline output.bin:1]: Do we want these bin and the gltf committed?

File: e2e/playwright/export-snapshots/gltf-standard.png
Comment (Irev-Dev) [inline e2e/playwright/export-snapshots/gltf-standard.png:1]: Wooohooo

File: output.bin
Comment (lf94) [reply output.bin:1]: I have no idea because I'm not sure why we commit the other files :) Up to what you think

File: e2e/playwright/snapshot-tests.spec.ts
Comment (Irev-Dev) [inline e2e/playwright/snapshot-tests.spec.ts:284]: Un tested by maybe this should be something like
```suggestion
              return fsp.writeFile(`./e2e/playwright/export-snapshots/${file.name}`, data).then(() => file.name)
```
So that these file end up in export-snapshots and therefore gitignored along with all the other exports?

Comment (lf94) [reply e2e/playwright/snapshot-tests.spec.ts:284]: Ah is that the condition? Sounds good then -  let's see...

Comment (lf94) [reply e2e/playwright/snapshot-tests.spec.ts:284]: (I'll use `path` to properly compute that path)

File: output.bin
Comment (Irev-Dev) [reply output.bin:1]: Okay well I don't think it should be committed, see this related comment
https://github.com/KittyCAD/modeling-app/pull/1688/files#r1523801832

The context is that we were committing all the exports originally, just so we had a e2e test covering exports, and if the exports changed or died we see it as exporting bad files.
The problem with this is everytime David made the smallest change to any of the formats that didn't really effect the export materially, we'd get diffs,
So sending the export back to the CLI to get a screenshot is a nice middle ground to sanity check the exports.

Once we did this we started gitignoring the raw exports with the exception of gltf-standard, because the screenshots wouldn't work. 

Now that you fixed it, I think we should only have screen shots commited

Comment (Irev-Dev) [reply output.bin:1]: Instead of my suggestion of making the unzipping save to the `snapshot-exports` folder, you could instead change the `check for changes` in the playwright action to only stage pngs?

Comment (lf94) [reply output.bin:1]: I'm on it

**Comment by lf94** [comment]
@Irev-Dev can I smash that merge button

**Comment by lf94** [comment]
Had to rebase again - I guess everyone is pushing to main at the same time :upside_down_face: :smiling_face_with_tear:

**Review by Irev-Dev — APPROVED** [review]
Whoop, let's merge!

**Pull request merged**
